### ISSUE TITLE
openssh_keygen does not set permissions correctly on public key

### ISSUE THREAD

**Issue by boustrophedon** [ansible/ansible#61411]
##### SUMMARY
In the openssh_keypair module, `set_fs_attributes_if_different` is not called on the public key, so the owner and group do not get set on the .pub file containing the public key.

I haven't tested it or reasoned about the code, but this may only occur when trying to run the task when the public key already exists.

<https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/crypto/openssh_keypair.py#L231>
In this line, file_args only receives the private key file location.

Additionally, there are "isPublicKeyValid" and "isPrivateKeyValid", and in the private key one we can see here <https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/crypto/openssh_keypair.py#L254> that we check the permissions for the private key, but there is no corresponding check in the public key.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

openssh_keypair

##### ANSIBLE VERSION

```paste below
ansible 2.8.4
  config file = <path_inside_home_directory>/ansible.cfg
  configured module search path = ['<home_directory>/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.4 (default, Jul 16 2019, 07:12:58) [GCC 9.1.0]

```

##### CONFIGURATION

```paste below
DEFAULT_HOST_LIST(<path_inside_home_directory>/aws-remote-work/ansible.cfg) = ['<path_inside_home_directory>/aws-remote-work/hosts']

```

##### OS / ENVIRONMENT

host: Linux host-machine 5.2.9-arch1-1-ARCH #1 SMP PREEMPT Fri Aug 16 11:29:43 UTC 2019 x86_64 GNU/Linux
remote: Linux remote-machine 4.9.0-9-amd64 #1 SMP Debian 4.9.168-1+deb9u5 (2019-08-11) x86_64 GNU/Linux

##### STEPS TO REPRODUCE

I am running the playbook with --become, so that it's executing as root

```yaml
- name: create ssh keypair in user home directory
  openssh_keypair:
    path: "/home/{{user_name}}/.ssh/my_ssh_key"
    type: ecdsa
    owner: "{{user_name}}"
    group: "users"
    mode: "u=rw,g=,o=" 
```

##### EXPECTED RESULTS

my_ssh_key should have permissions like
`-rw------- 1 user_name  users`
my_ssh_key.pub should have the same permissions:
`-rw------- 1 user_name  users`

##### ACTUAL RESULTS

only the private key has the correct permissions, and the public key has permissions
`-rw-r--r-- 1 root root`

**Comment by felixfontein** [comment]
resolved_by_pr #61658

**Comment by boustrophedon** [comment]
Thank you!

**Comment by felixfontein** [comment]
@boustrophedon you're welcome! It might have been a little too late for this week's 2.8.5 release, but it will definitely show up in 2.9.0 and in 2.8.6.

**Comment by felixfontein** [comment]
@boustrophedon please see the discussion in #62037; it won't show up in 2.8.6 since we don't want to break people's playbooks in non-major versions. There is a non-zero chance that someone is relying on the current behavior of the module. But it will be included in 2.9.0 (with a porting guide entry to notify users that this behavior changes). Sorry for my wrong information above.
